integrations/pooltogether-v4/components/form-yield-source-prize-pool-deposit.tsx

```diff
@@ -0,0 +1,158 @@
+import { USDC_CONTRACT } from '@/actions/pooltogether-v4/utils/usdc-contract-list'
+import { Checkbox } from '@/components/ui/checkbox'
+
+export function FormDeposit() {
+  const [isChecked, setIsChecked] = useState(false)
+  const [approvalAmount, setApprovalAmount] = useState<BigNumber>(BigNumber.from(0))
+  const [submitDeposit, setSubmitDeposit] = useState<boolean>(false)
+  const [isValidAmount, setValidAmount] = useState<boolean>(true)
+
+  const { address } = useAccount()
+  const prizePoolAddress = useLoadContractFromChainId(PRIZE_POOL_CONTRACT)
+  const usdcAddress = useLoadContractFromChainId(USDC_CONTRACT)
+
+  const { data: decimals } = useErc20Decimals({ address: usdcAddress })
+  const POWER: any = decimals != undefined ? BigNumber.from(10).pow(decimals) : BigNumber.from(10).pow(6)
+
+  const userBalance = useUserBalanceDeposit()
+  const isApproved = useUsdcApproval(userBalance)
+
+  const [depositAmount, setDepositAmount] = useState('')
+  const debouncedDepositAmount = useDebounce((POWER as any) * Number(depositAmount), 500)
+
+  // @ts-ignore
+  const { data, write: depositToken } = usePoolTogetherPrizePoolDepositTo({
+    address: prizePoolAddress,
+    args: [address, debouncedDepositAmount],
+    enabled: isApproved && Boolean(debouncedDepositAmount),
+    overrides: {
+      gasLimit: BigNumber.from(750000),
+    },
+  })
+
+  const { isLoading } = useWaitForTransaction({
+    hash: data?.hash,
+  })
+
+  // @ts-ignore
+  const { data: approveData, write: approval } = useErc20Approve({
+    address: usdcAddress,
+    args: [prizePoolAddress, approvalAmount],
+    overrides: {
+      gasLimit: BigNumber.from(750000),
+    },
+  })
+
+  const { isLoading: loadApprove, isSuccess: successApprove } = useWaitForTransaction({
+    hash: approveData?.hash,
+  })
+
+  useEffect(() => {
+    isChecked ? setApprovalAmount(ethers.constants.MaxInt256) : setApprovalAmount(BigNumber.from(debouncedDepositAmount))
+  }, [isChecked])
+
+  useEffect(() => {
+    if (successApprove || submitDeposit) {
+      depositToken?.()
+      setSubmitDeposit(false)
+    }
+  }, [successApprove, submitDeposit])
+
+  const handleSubmit = (event: any) => {
+    event.preventDefault()
+    if (parseFloat(depositAmount) >= 2.0) {
+      if (!isApproved) {
+        approval?.()
+      } else {
+        setSubmitDeposit(true)
+      }
+    } else {
+      setValidAmount(false)
+    }
+  }
+  const handleChange = (event: any) => {
+    const value = event.target.value
+    const regex = /^[0-9]*\.?[0-9]*$/
+    if (regex.test(value)) {
+      setDepositAmount(value)
+      setValidAmount(true)
+      setApprovalAmount(BigNumber.from(value * POWER))
+    }
+  }
+
+  const handleAmount = () => {
+    setDepositAmount(userBalance.toString())
+    setApprovalAmount(BigNumber.from(userBalance * POWER))
+  }
+
+  return (
+    <>
+      <Form.Root className="FormRoot" onSubmit={handleSubmit}>
```

Comment:
What's the meaning of this `FormRoot` className? same for "FormField"

Reply:
Useless, it was coming from the form sample from Radix UI, Removed !